generator/ServiceClientGeneratorLib/Generators/SourceFiles/ServiceConfig.tt

```diff
@@ -115,5 +116,26 @@ namespace <#=this.Config.Namespace#>
         public <#=parameter.nativeType#> <#=parameter.name#> { get; set; }
 
 <#}#>
+<# if (this.Config.EndpointsRuleSet != null) { #>
+        /// <summary>
+        /// Returns the endpoint that will be used for a particular request.
+        /// </summary>
+        /// <param name="request">Request for the desired service operation.</param>
+        /// <returns>The resolved endpoint for the given request.</returns>
+        public override Amazon.Runtime.Endpoints.Endpoint DetermineServiceOperationEndpoint(AmazonWebServiceRequest request)
```

Comment:
There is some possible work coming up that might involve account based endpoints which would be tied with the `AWSCredentials`. Putting this logic in the service config would make that work hard/impossible because service config has no concept of `AWSCredentials`. 

Maybe instead of taking in `AmazonWebServiceRequest` as a parameter take in a new container class like `EndpointParameters` which has a property for the `AmazonWebServiceRequest` and we can add other properties like account id in the future.